src/Snippets/Docs/Telemetry.cs

```diff
@@ -134,4 +134,36 @@ internal sealed class CustomMeteringEnricher : MeteringEnricher
     }
 
     #endregion
+
+    public static void SeverityOverrides()
+    {
+        var services = new ServiceCollection();
+
+        #region telemetry-severity-override
+
+        services.AddResiliencePipeline("my-strategy", (builder, context) =>
+        {
+            // Create a new instance of telemetry options by using copy-constructor of the global ones.
+            // This ensures that common configuration is preserved.
+            var telemetryOptions = new TelemetryOptions(context.GetOptions<TelemetryOptions>());
+
+            telemetryOptions.SeverityProvider = ev =>
```

Comment:
Would this approach allow the users to override the severity of the `ExecutionAttempt` event as well?

Reply:
Yup, but the information about the strategy type the produces it is missing.

Reply:
I was asking this because of this issue: https://github.com/App-vNext/Polly/issues/1958

Reply:
Added `SeverityProviderArguments` that make this API more flexible and future-proof.